### PR TITLE
Add records support to traffic_ctl and traffic_top.

### PR DESCRIPTION
**Pull request by jpeach**

### CODE REVIEW

**Comment by zwoop** [comment]
I guess I have no concerns with this, other than that I don't see that it's actually doing anything? It's sort of hidden with all the refactoring around using the ATS config parser, and then the lib records initialization, but what exactly does that do / solve?

**Comment by jpeach** [comment]
> On Jun 12, 2017, at 6:08 AM, Leif Hedstrom <notifications@github.com> wrote:
> 
> I guess I have no concerns with this, other than that I don't see that it's actually doing anything? It's sort of hidden with all the refactoring around using the ATS config parser,
> 
There are 3 separate commits here that separate out the refactoring. Probably better to review the commits one at a time.
> and then the lib records initialization, but what exactly does that do / solve?
> 
This lets you override the location of the management socket to use a non-default path. You can now point traffic_ctl to a test instance of traffic_server.
>

**Comment by dragon512** [comment]
@jpeach This looks great. I will have our Intern give this drop a test run.
@chitianhao Can you down load this version and see if the fixes work correctly for you.

File: cmd/traffic_ctl/Makefile.am [hunk 1]
```diff
@@ -35,8 +36,13 @@ traffic_ctl_SOURCES = \
   traffic_ctl.cc
 
 traffic_ctl_LDADD = \
+  $(top_builddir)/lib/records/librecords_p.a \
+  $(top_builddir)/mgmt/libmgmt_p.la \
+  $(top_builddir)/iocore/eventsystem/libinkevent.a \
+  $(top_builddir)/proxy/shared/libUglyLogStubs.a \
```
Comment (dragon512) [inline cmd/traffic_ctl/Makefile.am]: This is sign we need to refactor the layout of the code.

**Comment by SolidWallOfCode** [comment]
I'm not sure I'd go with "looks good" but it seems reasonable.  The use of `UglyLogStubs` indicates we have some severe problems in our dependency graph we need to deal with, although that's outside the scope of this PR.

**Comment by jpeach** [comment]
> On Jun 12, 2017, at 8:04 AM, Alan M. Carroll <notifications@github.com> wrote:
> 
> I'm not sure I'd go with "looks good" but it seems reasonable. The use of UglyLogStubs indicates we have some severe problems in our dependency graph we need to deal with

I think "tangle" is the word you are looking for ;)

**Comment by jpeach** [comment]
Thanks @dragon512

**Pull request merged**